OneCommunityGlobal/HighestGoodNetworkApp
# Tim - Add badge summary modal to People Reports page

# Description

This PR integrates a non-editable badge summary modal into the People Reports page. The modal is activated with a **Show Badges** button that is grouped with the other buttons on the People Reports page.

<img width="711" alt="pr-badgesummary-1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/82f379a6-84e3-434c-a378-68d0554c6753">
<img width="429" alt="pr-badgesummary-2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/1c13d9c0-9da7-4c75-9d08-b87518ffa30e">

## Related PRS (if any):
N/A

## Main changes explained:
- Adds a **Show Badges** button next to the current buttons
- Clicking the **Show Badges** button activates a static version of the **Badge Summary** modal found in **View Profile > Featured Badges > Select Featured**
- Editable fields and PDF options have been removed
- Adds a **Close** button to the modal
- Styling added/updated for desktop and tablet versions of the modal
- Adds new files to FE codebase: **BadgeSummaryVIz.jsx** and **BadgeSummaryViz.css**

## How to test:
1. `git checkout Tim_impl_badge_summary_on_peoples_report` to check into current branch
2. run `npm install` to ensure your modules are up to date (no new modules are added in this PR) 
3. `npm run start:local` to run this PR locally
4. log in as an admin user
5. go to **Reports → Reports → People Reports → Choose Person**
6. Select different users to test the modal with varying badge counts. To test a user with many badges use "Tim Volunteer". To test a user with zero badges use "Tim Mentor". Note also that as an admin you can assign new badges to users.
7. Test desktop ( > 1024 px) and tablet (<= 1024px) modal styling.
8. Verify the **Badge Summary** button and modal are working as intended

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/273cc258-0ce3-4ad4-91db-e3181b5c59cb

## Notes/Known Issues:
- Only Admin and Owner accounts should be able access the People Reports section of the app.
- There is a slight shift in the background screen when the modal is activated. This is also noticeable when activated the user profile version of the modal.
- There are various key prop and html errors on the People Reports page that are unrelated to this PR.
- Due to existing styling of the People Reports buttons, the **Show Badges** button wraps to a new line at width <= 787 px.
